web/services/models.py

```diff
@@ -73,15 +73,22 @@ def notify_report_received(sender, instance, signal, *args, **kwargs):
         # Don't send a notification if there was not email provided as we can't
         # actively do anything about it
         return
-
-    send_notification_to_slack(instance.user.name,
+    notification_thread = threading.Thread(target=send_notification_to_slack, args=(instance.user.name,
```

Comment:
Is this safe given that the method returns and doesn't keep the thread object alive? Might it be an issue if the `send_notification_to_slack` were to take a little longer than expected?

Reply:
I have looked into this and I think unless you need your main thread to wait on the result of the threading operation you are ok to create a thread which will clean up after itself afterwards.

I have added a sleep(60) to the threading task and tried running a script which sends 8 https requests in quick succession and the server seemed to be able to cope with that.

Reply:
There is also the question of whether the python thread correctly cleans up after itself and frees resouces when used in this way. My reading of the docs is that it will but online answers seems a bit inconsistent on this point.

Reply:
I was less worried about the `Thread` cleaning up after itself than the object being deleted while the thread was running because the reference count would drop to 0 when the function returns. You're test would suggest that it is okay but having my C++ hat on and wondering about object lifetime makes me cautious.

Reply:
I just tried
```python
import gc
import threading
import time

def run():
    print("start")
    time.sleep(25)
    print("end")

def start_thread():
    thread = threading.Thread(target=run)
    thread.start()
    print(gc.get_referrers(thread))

start_thread()
print("started")
```
and the referrers list shows multiple references, one being the main thread, so there seems to be no issue as you say :+1: